core/src/test/scala/io/finch/request/ApplicativeRequestReaderSpec.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2014, by Vladimir Kostyukov and Contributors.
+ *
+ * This file is a part of a Finch library that may be found at
+ *
+ *      https://github.com/finagle/finch
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributor(s):
+ * Jens Halm
+ */
+package io.finch.request
+
+import org.scalatest.{FlatSpec,Matchers}
+
+import com.twitter.finagle.httpx.Request
+import com.twitter.util.{Await,Throw}
+
+import io.finch.applicative._
+
+class ApplicativeRequestReaderSpec extends FlatSpec with Matchers {
+
+  
+  val reader: RequestReader[(Int,Double,Int)] = 
+    (RequiredIntParam("a") ~
+     RequiredDoubleParam("b") ~
+     RequiredIntParam("c")) { 
+       (a,b,c) => (a,b,c)
```

Comment:
Do you think we can avoid definition of `CanBuildX` and let users pattern match the result instead? Like in `Router`:

```scala
val reader: RequestReader[(Int,Double,Int)] = 
  (RequiredIntParam("a") ~
   RequiredDoubleParam("b") ~
   RequiredIntParam("c")) { case a ~ b ~ c => (a,b,c) }
```

I think this is a reasonable syntax.